Source/WebCore/rendering/BreakLines.h

```diff
@@ -130,55 +130,55 @@ inline unsigned nextBreakablePositionKeepingAllWords(const CharacterType* string
         if (string[i] == ideographicSpace)
             return i + 1;
     }
-    return length;
+    return string.size();
 }
 
 inline unsigned nextBreakablePositionKeepingAllWords(CachedLineBreakIteratorFactory& lineBreakIteratorFactory, unsigned startPosition)
 {
     auto stringView = lineBreakIteratorFactory.stringView();
     if (stringView.is8Bit())
-        return nextBreakablePositionKeepingAllWords<LChar, NonBreakingSpaceBehavior::TreatNonBreakingSpaceAsBreak>(stringView.characters8(), stringView.length(), startPosition);
-    return nextBreakablePositionKeepingAllWords<UChar, NonBreakingSpaceBehavior::TreatNonBreakingSpaceAsBreak>(stringView.characters16(), stringView.length(), startPosition);
+        return nextBreakablePositionKeepingAllWords<LChar, NonBreakingSpaceBehavior::TreatNonBreakingSpaceAsBreak>(stringView.span8(), startPosition);
+    return nextBreakablePositionKeepingAllWords<UChar, NonBreakingSpaceBehavior::TreatNonBreakingSpaceAsBreak>(stringView.span16(), startPosition);
 }
 
 inline unsigned nextBreakablePositionKeepingAllWordsIgnoringNBSP(CachedLineBreakIteratorFactory& lineBreakIteratorFactory, unsigned startPosition)
 {
     auto stringView = lineBreakIteratorFactory.stringView();
     if (stringView.is8Bit())
-        return nextBreakablePositionKeepingAllWords<LChar, NonBreakingSpaceBehavior::IgnoreNonBreakingSpace>(stringView.characters8(), stringView.length(), startPosition);
-    return nextBreakablePositionKeepingAllWords<UChar, NonBreakingSpaceBehavior::IgnoreNonBreakingSpace>(stringView.characters16(), stringView.length(), startPosition);
+        return nextBreakablePositionKeepingAllWords<LChar, NonBreakingSpaceBehavior::IgnoreNonBreakingSpace>(stringView.span8(), startPosition);
+    return nextBreakablePositionKeepingAllWords<UChar, NonBreakingSpaceBehavior::IgnoreNonBreakingSpace>(stringView.span16(), startPosition);
```

Comment:
I wonder if `nextBreakablePositionKeepingAllWords` and friends could just take a `span` that was a `subspace` of the `stringView` starting at `startPosition`.

Reply:
This applies to many places in this file!

Reply:
Actually, I guess not because we want the return value to be based on the complete `span`. So never mind!

Reply:
I think this is all dead code now.

Reply:
> I think this is all dead code now.

Oh! I'll check what's dead and update the PR to drop the dead code then. Thanks for pointing this out.

Reply:
TextUtil::findNextBreakablePosition() seems to call it?

Reply:
my bad. I must have confused it with some similar looking function. Sorry about that.